evm/sequencer/server.go

```diff
@@ -135,3 +111,27 @@ func (s *Sequencer) RegisterGameShard(
 	s.tq.AddInitMsg(req.GetNamespace(), req.GetRouterAddress())
 	return &shard.RegisterGameShardResponse{}, nil
 }
+
+// serverCallInterceptor catches calls to handlers and ensures they have the right secret routerKey.
+func (s *Sequencer) serverCallInterceptor(
+	ctx context.Context,
+	req any,
+	_ *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (resp any, err error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Errorf(codes.Unauthenticated, "missing metadata")
+	}
+
+	routerKey, ok := md["router-routerKey"]
```

Comment:
What about putting this "router-routerKey" as a constant in the evm/utils package that both the sequencer and router packages import?

Reply:
done. placed it in the rift package.